src/main/java/de/metas/ui/web/letter/LetterRestController.java

```diff
@@ -134,10 +149,53 @@ public JSONLetter getLetter(@PathVariable("letterId") final String letterId)
 		return JSONLetter.of(letter);
 	}
 
+	@Builder(builderMethodName = "fromLetterBuilder")
+	private I_C_Letter createPersistentLetter(final String subject, final String content)
+	{
+		final I_C_Letter persistentLetter = InterfaceWrapperHelper.newInstance(I_C_Letter.class);
+		persistentLetter.setLetterSubject(subject);
+		persistentLetter.setLetterBody(content);
+		persistentLetter.setLetterBodyParsed(content);
+		InterfaceWrapperHelper.save(persistentLetter);
+		return persistentLetter;
+	}
+
+	private I_C_Letter updatePersistentLetter(final WebuiLetter letter)
+	{
+		Check.assume(letter.getPersistentLetterId() > 0, "Letter ID should be > 0");
+		final Properties ctx = Env.getCtx();
+		final int C_BPartner_ID = Env.getContextAsInt(ctx, I_C_Letter.COLUMNNAME_C_BPartner_ID);
```

Comment:
i guess u will NEVER have C_BPartner_ID in global context.
Same for the other fields.

Reply:
also local variable names starting with upper case is not ok.